### ISSUE TITLE
getattrinfo: ENOTFOUND when using libnss plugins

### ISSUE THREAD

**Issue by k3rni** [beekeeper-studio/beekeeper-studio#55]
Setup: I'm using docker-compose to launch a Postgres instance, and [docker-nss](https://github.com/danni/docker-nss) to resolve its internal IP address by name, which is equivalent to its container name suffixed with `.docker`.

This works very well with `psql` and any database-connecting libraries (`libpq` wrappers for various languages). But beekeeper shows a red `getaddrinfo ENOTFOUND` when I enter such a host in the connect dialog. It does connect when I enter the IP addr directly, which is a workaround, but should work ootb.

**Comment by rathboma** [comment]
That's interesting. Yeah I'm using the default node-psql library without the libpq plug-in. I wonder if it does hostname resolution differently.

Can you show me how to set up that specific DNS setup?

**Comment by k3rni** [comment]
Sure! With one correction: there are two similar projects on Github, and I've linked to the wrong one, now corrected. This is all under Linux, other platforms may lack this feature. 

However, there should be nothing for you to do actually, if you're using regular libc name lookup, and not something that uses DNS directly (skipping other mechanisms, which this is one of).

1. Install `libnss-docker` (Debians), `nss-docker-git` (Arch), or just compile from source.
2. If not done so already by package manager, ensure you have it inserted into `/etc/nsswitch.conf` as instructed on the project page. On my system, the relevant line looks like this:
    
       hosts: files mymachines docker mdns4_minimal [NOTFOUND=return] resolve [!UNAVAIL=return] dns mdns4 myhostname
3. Launch a postgres instance in docker

       docker run -it --rm --name bk-psql -e POSTGRES_PASSWORD=helloworld postgres:alpine
4. From now on you can ping `bk-psql.docker` or connect to it with `psql -h bk-psql.docker -U 
postgres`

**Comment by rathboma** [comment]
I think this is a limitation with the Snap version of Beekeeper. I bet if you tried the AppImage it would work. I've had lots of issues with the Snap version -- lots of security walls

**Comment by k3rni** [comment]
You are correct, it does work indeed in the AppImage version. I think we can close this, maybe with a couple lines in the docs about the snap version being more restricted.
